sweep/sweeper.go

```diff
@@ -1566,17 +1536,33 @@ func (s *UtxoSweeper) updateSweeperInputs() pendingInputs {
 // sweepPendingInputs is called when the ticker fires. It will create clusters
 // and attempt to create and publish the sweeping transactions.
 func (s *UtxoSweeper) sweepPendingInputs(inputs pendingInputs) {
-	// We'll attempt to cluster all of our inputs with similar fee rates.
-	// Before attempting to sweep them, we'll sort them in descending fee
-	// rate order. We do this to ensure any inputs which have had their fee
-	// rate bumped are broadcast first in order enforce the RBF policy.
-	inputClusters := s.cfg.Aggregator.ClusterInputs(inputs)
-
-	for _, cluster := range inputClusters {
-		err := s.sweepCluster(cluster)
-		if err != nil {
-			log.Errorf("input cluster sweep: %v", err)
+	// Execute the sweep within a coin select lock. Otherwise the coins
+	// that we are going to spend may be selected for other transactions
+	// like funding of a channel.
+	//
+	// TODO(yy): decrease the lock scope - we need to remove the wallet
+	// used here, which means we need to ask the aggregator to return input
+	// sets and specifying whether wallet utoxs are needed or not. Then, by
+	// calling `TxInput.NeedWalletInput`, we can then lock and add the
+	// wallet input, creating a much smaller lock scope.
+	err := s.cfg.Wallet.WithCoinSelectLock(func() error {
+		// Cluster all of our inputs based on the specific Aggregator.
+		inputSets := s.cfg.Aggregator.ClusterInputs(
+			s.cfg.Wallet, inputs,
+		)
```

Comment:
🫡 Ok this cleans things up for me as it removes the concept of a Cluster as something separate from an InputSet.

Reply:
yeah sorry for putting it in your head first - could not find another way to provide this step-to-step change, otherwise it'd end up in a large commit.